crates/engine/tree/src/tree/persistence_state.rs

```diff
@@ -50,3 +74,18 @@ impl PersistenceState {
             BlockNumHash::new(last_persisted_block_number, last_persisted_block_hash);
     }
 }
+
+/// The currently running persistence action.
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub(crate) enum CurrentPersistenceAction<N: NodePrimitives> {
+    /// The persistence task is saving blocks.
+    SavingBlocks {
+        /// The blocks being saved
+        blocks: Vec<ExecutedBlock<N>>,
```

Comment:
Do we need to store here a vector of full `ExecutedBlock`? looks like we only use their `num_hash`

Reply:
also, do we need the `Vec`? the highest `BlockNumHash` could be enough

Reply:
right, I think the highest block num hash would work

Reply:
using this now